### PR TITLE
adds support for adding subscription preferences

### PR DESCRIPTION
**Pull request by suhasdeshpande**
## Description of the change

adds support for updating subscription preferences. 

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -130,6 +135,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 ## v1.0.1 - 2019-07-12
 
 [unreleased]: https://github.com/trycourier/courier-node/compare/v1.7.2...HEAD
```
Comment (tk26) [inline CHANGELOG.md:137]: nitpick - v1.7.3...HEAD

File: src/preferences/types.ts [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
-  categories: {
-    [notificationId: string]: {
-      status: string;
-    };
-  };
-  notifications: {
-    [notificationId: string]: {
-      status: string;
-    };
+export interface ICourierNotificationPreferences {
```
Comment (tk26) [inline src/preferences/types.ts:1]: I like this

File: CHANGELOG.md (hunk 1 above)
Comment (suhasdeshpande) [inline CHANGELOG.md:137]:
```suggestion
[unreleased]: https://github.com/trycourier/courier-node/compare/v1.7.3...HEAD
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -130,6 +135,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+[v1.7.3]: https://github.com/trycourier/courier-node/compare/v1.7.3...v1.7.2
```
Comment (suhasdeshpande) [inline CHANGELOG.md:138]:
```suggestion
[v1.7.3]: https://github.com/trycourier/courier-node/compare/v1.7.2...v1.7.3
```

**Review by tk26 — APPROVED** [review]

**Review by aydrian — COMMENTED** [review]
@scarney81 Does this need to be part of the 2.0 release or can we put it out as 1.8. Do you see anything as breaking? It looks like the added option to the subscribe method is optional so it shouldn't break anything.

File: CHANGELOG.md [hunk 4]
```diff
@@ -135,7 +139,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## v1.0.1 - 2019-07-12
 
-[unreleased]: https://github.com/trycourier/courier-node/compare/v1.7.3...HEAD
+[unreleased]: https://github.com/trycourier/courier-node/compare/v2.0.0...HEAD
+[v2.0.0]: https://github.com/trycourier/courier-node/compare/v2.0.0...v1.7.3
```
Comment (suhasdeshpande) [inline CHANGELOG.md:143]:
```suggestion
[v2.0.0]: https://github.com/trycourier/courier-node/compare/v1.7.3...v2.0.0
```

**Review by tk26 — APPROVED** [review]

**Pull request merged**